File: src/gui/pane/MessagePane.java
package gui.pane;

import gui.objects.Text;
import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.geometry.VPos;
import javafx.scene.control.ScrollPane;
import javafx.scene.layout.StackPane;
import javafx.scene.text.TextAlignment;
import static util.graphics.Util.*;

public class MessagePane extends OverlayPane<String> {

	private final Text text = new Text();

	public MessagePane() {
		text.setTextOrigin(VPos.CENTER);
		text.setTextAlignment(TextAlignment.JUSTIFY);
		setMinPrefMaxSize(text, -1.0);

		ScrollPane textPane = layScrollVTextCenter(text);
		textPane.setPrefWidth(400);
		textPane.setMaxWidth(400);

		StackPane root = layStack(textPane, Pos.CENTER);
		root.setPadding(new Insets(50));
		root.setMaxHeight(200);

		setContent(root);
	}

	@Override
	public void show(String message) {
		text.setText(message);
		super.show();
	}
}

File: src/gui/pane/MessagePane.kt
package gui.pane

import gui.objects.Text
import javafx.geometry.Insets
import javafx.geometry.Pos
import javafx.geometry.VPos
import javafx.scene.text.TextAlignment
import util.graphics.Util.layScrollVTextCenter
import util.graphics.Util.layStack
import util.graphics.setMinPrefMaxSize

class MessagePane: OverlayPane<String>() {

    private val text: Text

    init {
        text = Text().apply {
            textOrigin = VPos.CENTER
            textAlignment = TextAlignment.JUSTIFY
            setMinPrefMaxSize(-1.0)
        }

        val textPane = layScrollVTextCenter(text).apply {
            prefWidth = 400.0
            maxWidth = 400.0
        }

        content = layStack(textPane, Pos.CENTER).apply {
            padding = Insets(50.0)
            maxHeight = 200.0
        }
    }

    override fun show(data: String?) {
        text.text = data
        super.show()
    }

}